Task Liverpool/src/main/java/pages/SearchResultsPage.java

```diff
@@ -0,0 +1,152 @@
+    private By sortBySold = By.cssSelector(".col-lg-9 .dropdown-menu [datahref=\"/tienda/sold|1\"]");
+    private By productPrice = By.cssSelector(".m-product__card .a-card-discount");
+    private By rightArrow = By.cssSelector(".page-link .icon-arrow_right");
+    private By filterLowerThan$500Button = By.id("variants.prices.sortPrice-100-500");
+    //private By filterLowerThan$500Button2 = By.xpath("//input[@id='variants.prices.sortPrice-100-500']"); //Please ignore these locators, I am leaving them here for future reference
+    //private By filterLowerThan$500Button3 = By.cssSelector("#variants\\.prices\\.sortPrice-100-500"); //Please ignore these locators, I am leaving them here for future reference
+    private By noResultsText = By.cssSelector(".a-headline__noResults");
+    private By firstResult;
+    List<WebElement> pricesList;
+    List<Integer> pricesIntList;
+
+    public enum SortType {
+        RELEVANCE, LOWPRICE, HIGHPRICE, RATING, VIEWED, SOLD
+    }
+
+    public SearchResultsPage(WebDriver webDriver) {
+        super(webDriver);
+    }
+
+    public int getNumberOfResultsFromUpperRightLabel(){
+        return Integer.parseInt(getElementText(numberOfResultsFromUpperRightLabel));
+    }
+
+    //Logic for clicking on the Results Page Right Arrow
+    public boolean clickNextPage(){
+        if (isElementDisplayed(rightArrow)){
+            String productId = getElementAttribute(findElement(searchResultsDisplayed), "data-prodid");
+            firstResult = By.cssSelector("li.m-product__card[data-prodid=\"" + productId + "\"]");
+            scrollToElement(rightArrow);
+            click(rightArrow);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public int getTotalNumberOfResultsDisplayed() {
+        int totalResults = findElements(searchResultsDisplayed).size();
+
+        while (clickNextPage()){
+            totalResults += findElementsAfterACurrentElementDisappear(searchResultsDisplayed, firstResult).size();
+        }
+        return totalResults;
+    }
+
+    public void clickListViewButton(){
+        click(listViewButton);
+    }
+
+    public void clickSortProductsBy (SortType sortType) throws InterruptedException {
+
+        click(sortByDropDownList);
+        switch (sortType) {
+            case RELEVANCE:
+                click(sortByRelevance);
+                break;
+
+            case LOWPRICE:
+                click(sortByLowPrice);
+                break;
+
+            case HIGHPRICE:
+                click(sortByHighPrice);
+                break;
+
+            case RATING:
+                click(sortByRating);
+                break;
+
+            case VIEWED:
+                click(sortByViewed);
+                break;
+
+            case SOLD:
+                click(sortBySold);
+                break;
+        }
+        Thread.sleep(1000);
```

Comment:
avoid Thread.sleep

Reply:
I created a new method in BasePage called findElementsAfterAttributeChange where I wait until the attribute href for the sortByDropDownList locator contains a string that indicates how the products are being sorted. I had to make this because this attribute was the only thing I could find being updated at the same time in the DOM as the products in the screen after selecting how to Sort them.